src/Console/Command/SelfUpdate.php

```diff
@@ -0,0 +1,224 @@
+            $oldVersion = $updater->getOldVersion();
+        
+            if ($result) {
+                $this->output->writeln('<fg=green>PHPMND has been updated.</fg=green>');
+                $this->output->writeln(sprintf(
+                    '<fg=green>Current version is:</fg=green> <options=bold>%s</options=bold>.',
+                    $newVersion
+                ));
+                $this->output->writeln(sprintf(
+                    '<fg=green>Previous version was:</fg=green> <options=bold>%s</options=bold>.',
+                    $oldVersion
+                ));
+            } else {
+                $this->output->writeln('<fg=green>PHPMND is currently up to date.</fg=green>');
+                $this->output->writeln(sprintf(
+                    '<fg=green>Current version is:</fg=green> <options=bold>%s</options=bold>.',
+                    $oldVersion
+                ));
+            }
+        } catch (\Exception $e) {
+            $this->output->writeln(sprintf('Error: <fg=yellow>%s</fg=yellow>', $e->getMessage()));
+        }
+        $this->output->write(PHP_EOL);
+    }
+
+    /**
+     * Attempt to rollback to the previous phar version.
+     */
+    protected function rollback()
+    {
+        $updater = new Updater(null, false);
+        try {
+            $result = $updater->rollback();
+            if ($result) {
+                $this->output->writeln('<fg=green>PHPMND has been rolled back to prior version.</fg=green>');
+            } else {
+                $this->output->writeln('<fg=red>Rollback failed for reasons unknown.</fg=red>');
+            }
+        } catch (\Exception $e) {
+            $this->output->writeln(sprintf('Error: <fg=yellow>%s</fg=yellow>', $e->getMessage()));
+        }
+    }
+
+    protected function printAvailableUpdates()
+    {
+        $this->printCurrentLocalVersion();
+        $this->printCurrentStableVersion();
+    }
+
+    /**
+     * Print the current version of the phar in use.
+     */
+    protected function printCurrentLocalVersion()
+    {
+        $this->output->writeln(sprintf(
+            'Your current local build version is: <options=bold>%s</options=bold>',
+            $this->version
+        ));
+    }
+
+    /**
+     * Send updater to version printer.
+     */
+    protected function printCurrentStableVersion()
+    {
+        $this->printVersion($this->getStableUpdater());
+    }
+
+    /**
+     * Print a remotely available version.
+     * @param  Updater $updater
+     */
+    protected function printVersion(Updater $updater)
+    {
+        $stability = 'stable';
+        try {
+            if ($updater->hasUpdate()) {
+                $this->output->writeln(sprintf(
+                    'The current %s build available remotely is: <options=bold>%s</options=bold>',
+                    $stability,
+                    $updater->getNewVersion()
+                ));
+            } elseif (false == $updater->getNewVersion()) {
```

Comment:
Unfortunately, it does. There's an implementation shortfall in the phar-updater library, where this avoids calling Packagist twice. Something to be fixed there in a future version.